wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewController.swift

```diff
@@ -138,10 +138,13 @@ final class ConversationGuestOptionsViewController: UIViewController,
         _ viewModel: ConversationGuestOptionsViewModel,
         didUpdateState state: ConversationGuestOptionsViewModel.State
     ) {
-        tableView.reloadData()
-
-        (navigationController as? SpinnerCapableViewController)?.isLoadingViewVisible = state.isLoading
+        if state.isLoading {
+            activityIndicator.start()
```

Comment:
Considering how `BlockingActivityIndicator` is used in practice I'm wondering if its API should change to something like the following as it feels like the direction we are moving in swift is reacting to state changes and the current API is less declarative.

```swift
activityIndicator.update(isActive: state.isLoading)
```

Reply:
Very good point, thanks!
Added a method: 4a9448162c3119cb1d710abc26e40e6ebe655cc1